assets/model_monitoring/components/src/model_data_collector_preprocessor/genai_run.py

```diff
@@ -35,18 +39,51 @@ def _get_important_field_mapping() -> dict:
     return map
 
 
+def _drop_promoted_fields(df: DataFrame, promoted_fields_mapping: dict) -> DataFrame:
+    """Drop the promoted fields from dataframe to avoid data duplication and save storage space."""
+    def try_drop_field(df: DataFrame, col_name: str, field_name: str):
+        """Drop field from nested columns like context, attributes.
+
+        Will set column to null if dropping last field in the column.
```

Comment:
Yes. Mainly because if I just try dropping the last field in a column then an error is thrown and so I had to handle the error. The logic after catching the exception is up to us and we decided to keep a null column instead of removing the column